kola/tests/crio/crio.go

```diff
@@ -0,0 +1,373 @@
+		// Wait until listener is ready before trying anything
+		for {
+			_, err := c.SSH(dest, "sudo netstat -tulpn|grep 9988")
+			if err == nil {
+				break // socket is ready
+			}
+
+			exit, ok := err.(*ssh.ExitError)
+			if !ok || exit.Waitmsg.ExitStatus() != 1 { // 1 is the expected exit of grep -q
+				return err
+			}
+
+			select {
+			case <-ctx.Done():
+				return fmt.Errorf("timeout waiting for server")
+			default:
+				time.Sleep(100 * time.Millisecond)
+			}
+		}
+		podID := c.MustSSH(src, fmt.Sprintf("sudo crictl runp %s", crioConfigPod))
+		containerID := c.MustSSH(src, fmt.Sprintf("sudo crictl create %s %s %s",
+			podID, crioConfigContainer, crioConfigPod))
+
+		output := string(c.MustSSH(src, fmt.Sprintf("sudo crictl exec -t %s echo 'HELLO FROM CLIENT' | ncat %s 9988",
+			containerID, dest.PrivateIP())))
+		if output != "HELLO FROM SERVER" {
+			return fmt.Errorf(`unexpected result from listener: "%s"`, output)
+		}
+
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	if err := worker.Parallel(ctx, listener, talker); err != nil {
+		c.Fatal(err)
+	}
+}
+
+// crioNetworksReliably verifies that crio containers have a reliable network
+func crioNetworksReliably(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	crioConfigPod, crioConfigContainer, err := genContainer(
+		c, m, "ping", []string{"ping"},
+		[]string{"ping"})
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Here we generate 10 pods, each will run a container responsible for
+	// pinging to host
+	cmdCreatePod := fmt.Sprintf("sudo crictl runp %s", crioConfigPod)
+	output := ""
+	for x := 1; x <= 10; x++ {
+		podID := c.MustSSH(m, cmdCreatePod)
+		containerID := c.MustSSH(m, fmt.Sprintf("sudo crictl create %s %s %s",
+			podID, crioConfigContainer, crioConfigPod))
+		output = output + string(c.MustSSH(m, fmt.Sprintf("sudo crictl exec -t %s ping -i 0.2 172.17.0.1 -w 1 >/dev/null && echo PASS || echo FAIL", containerID)))
+	}
+
+	numPass := strings.Count(string(output), "PASS")
+
+	if numPass != 10 {
+		c.Fatalf("Expected 10 passes, but received %d passes with output: %s", numPass, output)
+	}
+
+}
+
+// getCrioInfo parses and returns the information crio provides via socket
+func getCrioInfo(c cluster.TestCluster, m platform.Machine) (simplifiedCrioInfo, error) {
+	target := simplifiedCrioInfo{}
+	crioInfoJSON, err := c.SSH(m, `sudo curl -s --unix-socket /var/run/crio/crio.sock http://crio/info`)
+
+	if err != nil {
+		return target, fmt.Errorf("could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(crioInfoJSON, &target)
+	if err != nil {
+		return target, fmt.Errorf("could not unmarshal info %q into known json: %v", string(crioInfoJSON), err)
+	}
+	return target, nil
+}
+
+// testCrioInfo test that crio info's output is as expected.
+func testCrioInfo(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	if _, err := c.SSH(m, `sudo systemctl start crio`); err != nil {
```

Comment:
`crio` is running by default, is this necessary? I spawned an instance and was able to `curl` the info without running this.

Reply:
If it's now on by default it can be removed.